pkg/controller/kfservice/kfservice_controller.go

```diff
@@ -132,16 +134,24 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	credentialReconciler := credentials.NewCredentialReconciler(r.Client)
+
 	serviceReconciler := ksvc.NewServiceReconciler(r.Client)
 	// Reconcile configurations
+
 	desiredDefault := resources.CreateKnativeConfiguration(constants.DefaultConfigurationName(kfsvc.Name),
 		kfsvc.ObjectMeta, &kfsvc.Spec.Default)
 
 	if err := controllerutil.SetControllerReference(kfsvc, desiredDefault, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	defaultConfiguration, err := serviceReconciler.ReconcileConfiguarion(context.TODO(), desiredDefault)
+	if err := credentialReconciler.ReconcileServiceAccount(context.TODO(), request.Namespace, kfsvc.Spec.Default.ServiceAccountName,
```

Comment:
This could follow the resources.CreateKnativeConfiguration pattern as  resources.CreateSecretVolumeAndEnv. Don't worry about it to much, though, just an observation.

Reply:
I have reorganized the files and renamed the credential reconciler to builder, wanna take a look at again ?